2025-10-09 03:04:34 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "MIGRATION_MODEL" not found (this database is empty); SQL statement:
SELECT ID, VERSION FROM MIGRATION_MODEL ORDER BY UPDATE_TIME DESC [42104-220]
2025-10-09 03:04:37 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "DATABASECHANGELOG" not found (this database is empty); SQL statement:
SELECT COUNT(*) FROM PUBLIC.DATABASECHANGELOG [42104-220]
2025-10-09 03:04:38 jdbc[4]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "DATABASECHANGELOGLOCK" not found (this database is empty); SQL statement:
SELECT COUNT(*) FROM PUBLIC.DATABASECHANGELOGLOCK [42104-220]
2025-10-09 03:04:38 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "DATABASECHANGELOG" not found; SQL statement:
SELECT COUNT(*) FROM PUBLIC.DATABASECHANGELOG [42102-220]
2025-10-09 05:30:17 database: CREATE CACHED TABLE "PUBLIC"."CLIENT"(
    "ID" CHARACTER VARYING(36) NOT NULL,
    "ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "FULL_SCOPE_ALLOWED" BOOLEAN DEFAULT FALSE NOT NULL,
    "CLIENT_ID" CHARACTER VARYING(255),
    "NOT_BEFORE" INTEGER,
    "PUBLIC_CLIENT" BOOLEAN DEFAULT FALSE NOT NULL,
    "SECRET" CHARACTER VARYING(255),
    "BASE_URL" CHARACTER VARYING(255),
    "BEARER_ONLY" BOOLEAN DEFAULT FALSE NOT NULL,
    "MANAGEMENT_URL" CHARACTER VARYING(255),
    "SURROGATE_AUTH_REQUIRED" BOOLEAN DEFAULT FALSE NOT NULL,
    "REALM_ID" CHARACTER VARYING(36),
    "PROTOCOL" CHARACTER VARYING(255),
    "NODE_REREG_TIMEOUT" INTEGER DEFAULT 0,
    "FRONTCHANNEL_LOGOUT" BOOLEAN DEFAULT FALSE NOT NULL,
    "CONSENT_REQUIRED" BOOLEAN DEFAULT FALSE NOT NULL,
    "NAME" CHARACTER VARYING(255),
    "SERVICE_ACCOUNTS_ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "CLIENT_AUTHENTICATOR_TYPE" CHARACTER VARYING(255),
    "ROOT_URL" CHARACTER VARYING(255),
    "DESCRIPTION" CHARACTER VARYING(255),
    "REGISTRATION_TOKEN" CHARACTER VARYING(255),
    "STANDARD_FLOW_ENABLED" BOOLEAN DEFAULT TRUE NOT NULL,
    "IMPLICIT_FLOW_ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "DIRECT_ACCESS_GRANTS_ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "ALWAYS_DISPLAY_IN_CONSOLE" BOOLEAN DEFAULT FALSE NOT NULL
)
org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "/opt/keycloak/data/h2/keycloakdb.mv.db"; SQL statement:
CREATE CACHED TABLE "PUBLIC"."CLIENT"(
    "ID" CHARACTER VARYING(36) NOT NULL,
    "ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "FULL_SCOPE_ALLOWED" BOOLEAN DEFAULT FALSE NOT NULL,
    "CLIENT_ID" CHARACTER VARYING(255),
    "NOT_BEFORE" INTEGER,
    "PUBLIC_CLIENT" BOOLEAN DEFAULT FALSE NOT NULL,
    "SECRET" CHARACTER VARYING(255),
    "BASE_URL" CHARACTER VARYING(255),
    "BEARER_ONLY" BOOLEAN DEFAULT FALSE NOT NULL,
    "MANAGEMENT_URL" CHARACTER VARYING(255),
    "SURROGATE_AUTH_REQUIRED" BOOLEAN DEFAULT FALSE NOT NULL,
    "REALM_ID" CHARACTER VARYING(36),
    "PROTOCOL" CHARACTER VARYING(255),
    "NODE_REREG_TIMEOUT" INTEGER DEFAULT 0,
    "FRONTCHANNEL_LOGOUT" BOOLEAN DEFAULT FALSE NOT NULL,
    "CONSENT_REQUIRED" BOOLEAN DEFAULT FALSE NOT NULL,
    "NAME" CHARACTER VARYING(255),
    "SERVICE_ACCOUNTS_ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "CLIENT_AUTHENTICATOR_TYPE" CHARACTER VARYING(255),
    "ROOT_URL" CHARACTER VARYING(255),
    "DESCRIPTION" CHARACTER VARYING(255),
    "REGISTRATION_TOKEN" CHARACTER VARYING(255),
    "STANDARD_FLOW_ENABLED" BOOLEAN DEFAULT TRUE NOT NULL,
    "IMPLICIT_FLOW_ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "DIRECT_ACCESS_GRANTS_ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "ALWAYS_DISPLAY_IN_CONSOLE" BOOLEAN DEFAULT FALSE NOT NULL
) [90028-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:168)
	at org.h2.mvstore.db.Store.createTable(Store.java:220)
	at org.h2.schema.Schema.createTable(Schema.java:797)
	at org.h2.command.ddl.CreateTable.update(CreateTable.java:112)
	at org.h2.engine.MetaRecord.prepareAndExecute(MetaRecord.java:77)
	at org.h2.engine.Database.executeMeta(Database.java:660)
	at org.h2.engine.Database.executeMeta(Database.java:634)
	at org.h2.engine.Database.<init>(Database.java:357)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.mvstore.MVStoreException: Reading from file sun.nio.ch.FileChannelImpl@3326e1f6 failed at 593109 (length -1), read 1536, remaining 0 [2.2.220/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:455)
	at org.h2.mvstore.FileStore.readFully(FileStore.java:719)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:60)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:28)
	at org.h2.mvstore.SFChunk.readFully(SFChunk.java:35)
	at org.h2.mvstore.Chunk.readBufferForPage(Chunk.java:465)
	at org.h2.mvstore.FileStore.readPage(FileStore.java:1961)
	at org.h2.mvstore.MVStore.readPage(MVStore.java:1022)
	at org.h2.mvstore.MVMap.readPage(MVMap.java:632)
	at org.h2.mvstore.MVMap.readOrCreateRootPage(MVMap.java:657)
	at org.h2.mvstore.MVMap.setRootPos(MVMap.java:642)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:479)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:426)
	at org.h2.mvstore.tx.TransactionStore.openMap(TransactionStore.java:564)
	at org.h2.mvstore.tx.TransactionStore.openVersionedMap(TransactionStore.java:550)
	at org.h2.mvstore.tx.Transaction.openMap(Transaction.java:456)
	at org.h2.mvstore.db.MVPrimaryIndex.<init>(MVPrimaryIndex.java:53)
	at org.h2.mvstore.db.MVTable.<init>(MVTable.java:156)
	at org.h2.mvstore.db.Store.createTable(Store.java:216)
	... 20 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.readInternal(FileChannelImpl.java:844)
	at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:824)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:441)
	... 38 more
2025-10-09 05:30:18 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: IO Exception: "/opt/keycloak/data/h2/keycloakdb.mv.db" [90028-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:168)
	at org.h2.mvstore.db.Store.createTable(Store.java:220)
	at org.h2.schema.Schema.createTable(Schema.java:797)
	at org.h2.command.ddl.CreateTable.update(CreateTable.java:112)
	at org.h2.engine.MetaRecord.prepareAndExecute(MetaRecord.java:77)
	at org.h2.engine.Database.executeMeta(Database.java:660)
	at org.h2.engine.Database.executeMeta(Database.java:634)
	at org.h2.engine.Database.<init>(Database.java:357)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "/opt/keycloak/data/h2/keycloakdb.mv.db"; SQL statement:
CREATE CACHED TABLE "PUBLIC"."CLIENT"(
    "ID" CHARACTER VARYING(36) NOT NULL,
    "ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "FULL_SCOPE_ALLOWED" BOOLEAN DEFAULT FALSE NOT NULL,
    "CLIENT_ID" CHARACTER VARYING(255),
    "NOT_BEFORE" INTEGER,
    "PUBLIC_CLIENT" BOOLEAN DEFAULT FALSE NOT NULL,
    "SECRET" CHARACTER VARYING(255),
    "BASE_URL" CHARACTER VARYING(255),
    "BEARER_ONLY" BOOLEAN DEFAULT FALSE NOT NULL,
    "MANAGEMENT_URL" CHARACTER VARYING(255),
    "SURROGATE_AUTH_REQUIRED" BOOLEAN DEFAULT FALSE NOT NULL,
    "REALM_ID" CHARACTER VARYING(36),
    "PROTOCOL" CHARACTER VARYING(255),
    "NODE_REREG_TIMEOUT" INTEGER DEFAULT 0,
    "FRONTCHANNEL_LOGOUT" BOOLEAN DEFAULT FALSE NOT NULL,
    "CONSENT_REQUIRED" BOOLEAN DEFAULT FALSE NOT NULL,
    "NAME" CHARACTER VARYING(255),
    "SERVICE_ACCOUNTS_ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "CLIENT_AUTHENTICATOR_TYPE" CHARACTER VARYING(255),
    "ROOT_URL" CHARACTER VARYING(255),
    "DESCRIPTION" CHARACTER VARYING(255),
    "REGISTRATION_TOKEN" CHARACTER VARYING(255),
    "STANDARD_FLOW_ENABLED" BOOLEAN DEFAULT TRUE NOT NULL,
    "IMPLICIT_FLOW_ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "DIRECT_ACCESS_GRANTS_ENABLED" BOOLEAN DEFAULT FALSE NOT NULL,
    "ALWAYS_DISPLAY_IN_CONSOLE" BOOLEAN DEFAULT FALSE NOT NULL
) [90028-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 23 more
Caused by: org.h2.mvstore.MVStoreException: Reading from file sun.nio.ch.FileChannelImpl@3326e1f6 failed at 593109 (length -1), read 1536, remaining 0 [2.2.220/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:455)
	at org.h2.mvstore.FileStore.readFully(FileStore.java:719)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:60)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:28)
	at org.h2.mvstore.SFChunk.readFully(SFChunk.java:35)
	at org.h2.mvstore.Chunk.readBufferForPage(Chunk.java:465)
	at org.h2.mvstore.FileStore.readPage(FileStore.java:1961)
	at org.h2.mvstore.MVStore.readPage(MVStore.java:1022)
	at org.h2.mvstore.MVMap.readPage(MVMap.java:632)
	at org.h2.mvstore.MVMap.readOrCreateRootPage(MVMap.java:657)
	at org.h2.mvstore.MVMap.setRootPos(MVMap.java:642)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:479)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:426)
	at org.h2.mvstore.tx.TransactionStore.openMap(TransactionStore.java:564)
	at org.h2.mvstore.tx.TransactionStore.openVersionedMap(TransactionStore.java:550)
	at org.h2.mvstore.tx.Transaction.openMap(Transaction.java:456)
	at org.h2.mvstore.db.MVPrimaryIndex.<init>(MVPrimaryIndex.java:53)
	at org.h2.mvstore.db.MVTable.<init>(MVTable.java:156)
	at org.h2.mvstore.db.Store.createTable(Store.java:216)
	... 20 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.readInternal(FileChannelImpl.java:844)
	at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:824)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:441)
	... 38 more
2025-10-09 05:30:18 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@3326e1f6 failed; length 8192 at 16384 [2.2.220/2]" [50000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1547)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1528)
	at org.h2.mvstore.FileStore.lambda$serializeAndStore$3(FileStore.java:1433)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1402)
	at org.h2.mvstore.FileStore.serializeAndStore(FileStore.java:1433)
	at org.h2.mvstore.FileStore.lambda$storeIt$2(FileStore.java:1377)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1402)
	at org.h2.mvstore.FileStore.storeIt(FileStore.java:1376)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:844)
	at org.h2.mvstore.MVStore.store(MVStore.java:809)
	at org.h2.mvstore.MVStore.commit(MVStore.java:778)
	at org.h2.mvstore.MVStore.commit(MVStore.java:770)
	at org.h2.mvstore.tx.TransactionStore.close(TransactionStore.java:365)
	at org.h2.mvstore.db.Store.close(Store.java:349)
	at org.h2.engine.Database.closeOpenFilesAndUnlock(Database.java:1263)
	at org.h2.engine.Database.<init>(Database.java:399)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@3326e1f6 failed; length 8192 at 16384 [2.2.220/2]" [50000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 34 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@3326e1f6 failed; length 8192 at 16384 [2.2.220/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:478)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:67)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:28)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:366)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:28)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1525)
	... 28 more
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:159)
	at java.base/sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:854)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:474)
	... 33 more
2025-10-09 05:30:19 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:334)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-09 05:30:24 database: CREATE CACHED TABLE "PUBLIC"."PROTOCOL_MAPPER"(
    "ID" CHARACTER VARYING(36) NOT NULL,
    "NAME" CHARACTER VARYING(255) NOT NULL,
    "PROTOCOL" CHARACTER VARYING(255) NOT NULL,
    "PROTOCOL_MAPPER_NAME" CHARACTER VARYING(255) NOT NULL,
    "CLIENT_ID" CHARACTER VARYING(36),
    "CLIENT_SCOPE_ID" CHARACTER VARYING(36)
)
org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "/opt/keycloak/data/h2/keycloakdb.mv.db"; SQL statement:
CREATE CACHED TABLE "PUBLIC"."PROTOCOL_MAPPER"(
    "ID" CHARACTER VARYING(36) NOT NULL,
    "NAME" CHARACTER VARYING(255) NOT NULL,
    "PROTOCOL" CHARACTER VARYING(255) NOT NULL,
    "PROTOCOL_MAPPER_NAME" CHARACTER VARYING(255) NOT NULL,
    "CLIENT_ID" CHARACTER VARYING(36),
    "CLIENT_SCOPE_ID" CHARACTER VARYING(36)
) [90028-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:168)
	at org.h2.mvstore.db.Store.createTable(Store.java:220)
	at org.h2.schema.Schema.createTable(Schema.java:797)
	at org.h2.command.ddl.CreateTable.update(CreateTable.java:112)
	at org.h2.engine.MetaRecord.prepareAndExecute(MetaRecord.java:77)
	at org.h2.engine.Database.executeMeta(Database.java:660)
	at org.h2.engine.Database.executeMeta(Database.java:634)
	at org.h2.engine.Database.<init>(Database.java:357)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.mvstore.MVStoreException: Reading from file sun.nio.ch.FileChannelImpl@15edaef7 failed at 580200 (length -1), read 48, remaining 0 [2.2.220/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:455)
	at org.h2.mvstore.FileStore.readFully(FileStore.java:719)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:60)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:28)
	at org.h2.mvstore.SFChunk.readFully(SFChunk.java:35)
	at org.h2.mvstore.Chunk.readBufferForPage(Chunk.java:465)
	at org.h2.mvstore.FileStore.readPage(FileStore.java:1961)
	at org.h2.mvstore.MVStore.readPage(MVStore.java:1022)
	at org.h2.mvstore.MVMap.readPage(MVMap.java:632)
	at org.h2.mvstore.MVMap.readOrCreateRootPage(MVMap.java:657)
	at org.h2.mvstore.MVMap.setRootPos(MVMap.java:642)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:479)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:426)
	at org.h2.mvstore.tx.TransactionStore.openMap(TransactionStore.java:564)
	at org.h2.mvstore.tx.TransactionStore.openVersionedMap(TransactionStore.java:550)
	at org.h2.mvstore.tx.Transaction.openMap(Transaction.java:456)
	at org.h2.mvstore.db.MVPrimaryIndex.<init>(MVPrimaryIndex.java:53)
	at org.h2.mvstore.db.MVTable.<init>(MVTable.java:156)
	at org.h2.mvstore.db.Store.createTable(Store.java:216)
	... 20 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.readInternal(FileChannelImpl.java:844)
	at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:824)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:441)
	... 38 more
2025-10-09 05:30:24 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: IO Exception: "/opt/keycloak/data/h2/keycloakdb.mv.db" [90028-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.mvstore.db.Store.convertMVStoreException(Store.java:168)
	at org.h2.mvstore.db.Store.createTable(Store.java:220)
	at org.h2.schema.Schema.createTable(Schema.java:797)
	at org.h2.command.ddl.CreateTable.update(CreateTable.java:112)
	at org.h2.engine.MetaRecord.prepareAndExecute(MetaRecord.java:77)
	at org.h2.engine.Database.executeMeta(Database.java:660)
	at org.h2.engine.Database.executeMeta(Database.java:634)
	at org.h2.engine.Database.<init>(Database.java:357)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "/opt/keycloak/data/h2/keycloakdb.mv.db"; SQL statement:
CREATE CACHED TABLE "PUBLIC"."PROTOCOL_MAPPER"(
    "ID" CHARACTER VARYING(36) NOT NULL,
    "NAME" CHARACTER VARYING(255) NOT NULL,
    "PROTOCOL" CHARACTER VARYING(255) NOT NULL,
    "PROTOCOL_MAPPER_NAME" CHARACTER VARYING(255) NOT NULL,
    "CLIENT_ID" CHARACTER VARYING(36),
    "CLIENT_SCOPE_ID" CHARACTER VARYING(36)
) [90028-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 23 more
Caused by: org.h2.mvstore.MVStoreException: Reading from file sun.nio.ch.FileChannelImpl@15edaef7 failed at 580200 (length -1), read 48, remaining 0 [2.2.220/1]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:455)
	at org.h2.mvstore.FileStore.readFully(FileStore.java:719)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:60)
	at org.h2.mvstore.SingleFileStore.readFully(SingleFileStore.java:28)
	at org.h2.mvstore.SFChunk.readFully(SFChunk.java:35)
	at org.h2.mvstore.Chunk.readBufferForPage(Chunk.java:465)
	at org.h2.mvstore.FileStore.readPage(FileStore.java:1961)
	at org.h2.mvstore.MVStore.readPage(MVStore.java:1022)
	at org.h2.mvstore.MVMap.readPage(MVMap.java:632)
	at org.h2.mvstore.MVMap.readOrCreateRootPage(MVMap.java:657)
	at org.h2.mvstore.MVMap.setRootPos(MVMap.java:642)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:479)
	at org.h2.mvstore.MVStore.openMap(MVStore.java:426)
	at org.h2.mvstore.tx.TransactionStore.openMap(TransactionStore.java:564)
	at org.h2.mvstore.tx.TransactionStore.openVersionedMap(TransactionStore.java:550)
	at org.h2.mvstore.tx.Transaction.openMap(Transaction.java:456)
	at org.h2.mvstore.db.MVPrimaryIndex.<init>(MVPrimaryIndex.java:53)
	at org.h2.mvstore.db.MVTable.<init>(MVTable.java:156)
	at org.h2.mvstore.db.Store.createTable(Store.java:216)
	... 20 more
Caused by: java.nio.channels.ClosedByInterruptException
	at java.base/java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:199)
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:171)
	at java.base/sun.nio.ch.FileChannelImpl.readInternal(FileChannelImpl.java:844)
	at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:824)
	at org.h2.mvstore.DataUtils.readFully(DataUtils.java:441)
	... 38 more
2025-10-09 05:30:24 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@15edaef7 failed; length 8192 at 16384 [2.2.220/2]" [50000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1547)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1528)
	at org.h2.mvstore.FileStore.lambda$serializeAndStore$3(FileStore.java:1433)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1402)
	at org.h2.mvstore.FileStore.serializeAndStore(FileStore.java:1433)
	at org.h2.mvstore.FileStore.lambda$storeIt$2(FileStore.java:1377)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1402)
	at org.h2.mvstore.FileStore.storeIt(FileStore.java:1376)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:844)
	at org.h2.mvstore.MVStore.store(MVStore.java:809)
	at org.h2.mvstore.MVStore.commit(MVStore.java:778)
	at org.h2.mvstore.MVStore.commit(MVStore.java:770)
	at org.h2.mvstore.tx.TransactionStore.close(TransactionStore.java:365)
	at org.h2.mvstore.db.Store.close(Store.java:349)
	at org.h2.engine.Database.closeOpenFilesAndUnlock(Database.java:1263)
	at org.h2.engine.Database.<init>(Database.java:399)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@15edaef7 failed; length 8192 at 16384 [2.2.220/2]" [50000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 34 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@15edaef7 failed; length 8192 at 16384 [2.2.220/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:478)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:67)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:28)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:366)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:28)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1525)
	... 28 more
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:159)
	at java.base/sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:854)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:474)
	... 33 more
2025-10-09 05:30:25 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:334)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-09 05:30:30 database: flush
org.h2.message.DbException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@325ffa57 failed; length 8192 at 16384 [2.2.220/2]" [50000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:407)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:122)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:1547)
	at org.h2.mvstore.MVStore.panic(MVStore.java:371)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1528)
	at org.h2.mvstore.FileStore.lambda$serializeAndStore$3(FileStore.java:1433)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1402)
	at org.h2.mvstore.FileStore.serializeAndStore(FileStore.java:1433)
	at org.h2.mvstore.FileStore.lambda$storeIt$2(FileStore.java:1377)
	at org.h2.mvstore.FileStore.submitOrRun(FileStore.java:1402)
	at org.h2.mvstore.FileStore.storeIt(FileStore.java:1376)
	at org.h2.mvstore.MVStore.storeNow(MVStore.java:844)
	at org.h2.mvstore.MVStore.store(MVStore.java:809)
	at org.h2.mvstore.MVStore.commit(MVStore.java:778)
	at org.h2.mvstore.MVStore.commit(MVStore.java:770)
	at org.h2.mvstore.tx.TransactionStore.close(TransactionStore.java:365)
	at org.h2.mvstore.db.Store.close(Store.java:349)
	at org.h2.engine.Database.closeOpenFilesAndUnlock(Database.java:1263)
	at org.h2.engine.Database.<init>(Database.java:399)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@325ffa57 failed; length 8192 at 16384 [2.2.220/2]" [50000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:566)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 34 more
Caused by: org.h2.mvstore.MVStoreException: Writing to sun.nio.ch.FileChannelImpl@325ffa57 failed; length 8192 at 16384 [2.2.220/2]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:996)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:478)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:67)
	at org.h2.mvstore.SingleFileStore.writeFully(SingleFileStore.java:28)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:366)
	at org.h2.mvstore.RandomAccessStore.writeChunk(RandomAccessStore.java:28)
	at org.h2.mvstore.FileStore.storeBuffer(FileStore.java:1525)
	... 28 more
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:159)
	at java.base/sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:854)
	at org.h2.mvstore.DataUtils.writeFully(DataUtils.java:474)
	... 33 more
2025-10-09 10:23:18 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:690)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.message.DbException.get(DbException.java:188)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1375)
	at org.h2.jdbcx.JdbcXAConnection$PooledJdbcConnection.checkClosed(JdbcXAConnection.java:473)
	at org.h2.jdbc.JdbcConnection.rollback(JdbcConnection.java:463)
	at org.h2.jdbcx.JdbcXAConnection$PooledJdbcConnection.close(JdbcXAConnection.java:453)
	at org.h2.jdbcx.JdbcXAConnection.close(JdbcXAConnection.java:76)
	at io.agroal.pool.ConnectionHandler.closeConnection(ConnectionHandler.java:185)
	at io.agroal.pool.ConnectionPool$DestroyConnectionTask.run(ConnectionPool.java:787)
	at io.agroal.pool.ConnectionPool.close(ConnectionPool.java:196)
	at io.agroal.pool.DataSource.close(DataSource.java:79)
	at io.quarkus.agroal.runtime.DataSources.stop(DataSources.java:454)
	at io.quarkus.agroal.runtime.DataSources_Bean.doDestroy(Unknown Source)
	at io.quarkus.agroal.runtime.DataSources_Bean.destroy(Unknown Source)
	at io.quarkus.agroal.runtime.DataSources_Bean.destroy(Unknown Source)
	at io.quarkus.arc.impl.AbstractInstanceHandle.destroyInternal(AbstractInstanceHandle.java:82)
	at io.quarkus.arc.impl.ContextInstanceHandleImpl.destroy(ContextInstanceHandleImpl.java:21)
	at io.quarkus.arc.impl.AbstractSharedContext.destroy(AbstractSharedContext.java:96)
	at io.quarkus.arc.impl.ArcContainerImpl.shutdown(ArcContainerImpl.java:468)
	at io.quarkus.arc.Arc.shutdown(Arc.java:66)
	at io.quarkus.arc.runtime.ArcRecorder$1.run(ArcRecorder.java:53)
	at io.quarkus.runtime.StartupContext.runAllInReverseOrder(StartupContext.java:84)
	at io.quarkus.runtime.StartupContext.close(StartupContext.java:73)
	at io.quarkus.runner.ApplicationImpl.doStop(Unknown Source)
	at io.quarkus.runtime.Application.stop(Application.java:208)
	at io.quarkus.runtime.Application.stop(Application.java:155)
	at io.quarkus.runtime.ApplicationLifecycleManager$ShutdownHookThread.run(ApplicationLifecycleManager.java:436)
2025-10-09 11:00:26 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:690)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.message.DbException.get(DbException.java:188)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1375)
	at org.h2.jdbcx.JdbcXAConnection$PooledJdbcConnection.checkClosed(JdbcXAConnection.java:473)
	at org.h2.jdbc.JdbcConnection.rollback(JdbcConnection.java:463)
	at org.h2.jdbcx.JdbcXAConnection$PooledJdbcConnection.close(JdbcXAConnection.java:453)
	at org.h2.jdbcx.JdbcXAConnection.close(JdbcXAConnection.java:76)
	at io.agroal.pool.ConnectionHandler.closeConnection(ConnectionHandler.java:185)
	at io.agroal.pool.ConnectionPool$DestroyConnectionTask.run(ConnectionPool.java:787)
	at io.agroal.pool.ConnectionPool.close(ConnectionPool.java:196)
	at io.agroal.pool.DataSource.close(DataSource.java:79)
	at io.quarkus.agroal.runtime.DataSources.stop(DataSources.java:454)
	at io.quarkus.agroal.runtime.DataSources_Bean.doDestroy(Unknown Source)
	at io.quarkus.agroal.runtime.DataSources_Bean.destroy(Unknown Source)
	at io.quarkus.agroal.runtime.DataSources_Bean.destroy(Unknown Source)
	at io.quarkus.arc.impl.AbstractInstanceHandle.destroyInternal(AbstractInstanceHandle.java:82)
	at io.quarkus.arc.impl.ContextInstanceHandleImpl.destroy(ContextInstanceHandleImpl.java:21)
	at io.quarkus.arc.impl.AbstractSharedContext.destroy(AbstractSharedContext.java:96)
	at io.quarkus.arc.impl.ArcContainerImpl.shutdown(ArcContainerImpl.java:468)
	at io.quarkus.arc.Arc.shutdown(Arc.java:66)
	at io.quarkus.arc.runtime.ArcRecorder$1.run(ArcRecorder.java:53)
	at io.quarkus.runtime.StartupContext.runAllInReverseOrder(StartupContext.java:84)
	at io.quarkus.runtime.StartupContext.close(StartupContext.java:73)
	at io.quarkus.runner.ApplicationImpl.doStop(Unknown Source)
	at io.quarkus.runtime.Application.stop(Application.java:208)
	at io.quarkus.runtime.Application.stop(Application.java:155)
	at io.quarkus.runtime.ApplicationLifecycleManager$ShutdownHookThread.run(ApplicationLifecycleManager.java:436)
2025-10-13 04:12:35 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:334)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:12:42 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:12:49 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:12:56 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:334)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:27:24 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:27:46 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:27:52 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:42:26 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:42:44 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:42:50 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:42:58 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:57:21 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:57:51 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 04:57:59 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 05:12:24 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 05:12:41 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 05:12:59 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 05:27:22 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 05:27:44 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:334)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
2025-10-13 05:27:57 database: opening /opt/keycloak/data/h2/keycloakdb
org.h2.message.DbException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.store.FileLock.getExceptionFatal(FileLock.java:429)
	at org.h2.store.FileLock.sleep(FileLock.java:424)
	at org.h2.store.FileLock.lockFile(FileLock.java:324)
	at org.h2.store.FileLock.lock(FileLock.java:110)
	at org.h2.engine.Database.<init>(Database.java:315)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLNonTransientConnectionException: Error opening database: "Sleep interrupted" [8000-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:516)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	... 20 more
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at org.h2.store.FileLock.sleep(FileLock.java:422)
	... 17 more
